src/main/java/com/artozersky/HackerNewsAPI/cache/impl/CacheEntityServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.artozersky.HackerNewsAPI.cache.impl;
+
+
+import com.artozersky.HackerNewsAPI.cache.CacheEntityService;
+import com.artozersky.HackerNewsAPI.model.impl.NewsPostModelImpl;
+
+import java.util.List;
+import java.time.LocalDateTime;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CacheEntityServiceImpl implements CacheEntityService {
+
+    private final CacheEntityImpl cacheEntity;
+
+    @Autowired
+    public CacheEntityServiceImpl(@Value("${cache.size:100}") int maxSize) {
+
+        this.cacheEntity = new CacheEntityImpl(maxSize);
+
+    }
+
+    @Override
+    public NewsPostModelImpl getPostFromCacheById(Long postId) {
+
+        return cacheEntity.get(postId);
+    }
+
+    @Override
+    public List<NewsPostModelImpl> getAllPostsFromCache() {
+
+        List<NewsPostModelImpl> cachedPosts = cacheEntity.getAll();
+
+        System.out.println("inside getAllPostsFromCache size : " + cachedPosts.size());
+
+        boolean isStale = cachedPosts.stream().anyMatch(post -> {
+            int currentElapsedTime = (int) java.time.Duration.between(post.getCreatedAt(), LocalDateTime.now()).toHours();
+            return Math.abs(post.getTimeElapsed() - currentElapsedTime) > 1;//more than one hour difference
+        });
+
+        if(isStale) {
+           this.clearCache();
+           System.out.println("Cache invalidated due to stale data");
```

Comment:
why system out? who can see this? who will actually look at this? you are trying to figure out if 1 hour was passed and then clear the cache, i think it is a better idea not to implement such a logic in this method because this is the most critical method and called the most time so every logic here is crucial in terms of performance. please implement an async functionality that updated all the time elapsed fields in the posts in the database every 1 hour (configurable by the user from the application.properties) and clears the cache also. now the logic should be if cache is empty bring from database if not bring from cache. simple and fast. then you will have a different thread that is waking up every 1 hour and doing async operations on the database and clearing the cache. be careful from getting from cache while you are deleting from it.

Reply:
fixed. now its logger, but i also need to erase it